10_rest/willshen8/pkg/puppy/syncstore.go

```diff
@@ -0,0 +1,76 @@
+package puppy
+
+import (
+	"strconv"
+	"sync"
+)
+
+type SyncStore struct {
+	syncStore sync.Map
+	nextID    uint32
+	mu        sync.Mutex
+}
+
+// NewMapStore initialise a new SyncStore
+func NewSyncStore() *SyncStore {
+	return &SyncStore{syncStore: sync.Map{}}
+}
+
+// CreatePuppy create a new puppy and store in mapStore.
+func (s *SyncStore) CreatePuppy(p *Puppy) (uint32, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if i, err := strconv.Atoi(p.Value); err == nil {
+		if i < 0 {
+			return 0, Errorf(ErrorInvalidInput, "Puppy value can't be negative.")
+		}
+	} else {
+		return 0, Errorf(ErrorInvalidInput, "Unrecognised input.")
+	}
```

Comment:
```go 
         i, err := strconv.Atoi(p.Value)
         if err != nil {
		return 0, Errorf(ErrorInvalidInput, "Puppy value can't be negative.")
	} 
         if i < 0 {
		return 0, Errorf(ErrorInvalidInput, "Unrecognised input")
	} 
```

Typically error messages don't end in periods for chainability.

Reply:
Done

Reply:
it's not done, or am i blind?

Reply:
I thought it's to do with commenting ending with period. Now fixed.